### ISSUE TITLE
Add testing support

### ISSUE THREAD

**Issue by brollb** [dfst/deepforge-caffe#112]
_Created by: brollb_

Architectures should have testing support

**Comment by brollb** [comment]
_Created by: brollb_

Caffe information: http://caffe.berkeleyvision.org/tutorial/interfaces.html

**Comment by brollb** [comment]
Added ~153450 label

**Comment by brollb** [comment]
I have the caffe testing running but I am getting a strange caffe error in my test example:

```
I0216 15:54:23.981935 13257 net.cpp:283] Network initialization done.
F0216 15:54:29.152992 13257 net.cpp:829] Cannot copy param 0 weights from layer 'ip1'; shape mismatch.  Source param shape is 500 186050 (93025000); target param shape is 500 8450 (4225000). To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.
*** Check failure stack trace: ***
    @     0x7fdf99790f9d  google::LogMessage::Fail()
    @     0x7fdf99792e03  google::LogMessage::SendToLog()
    @     0x7fdf99790b2b  google::LogMessage::Flush()
    @     0x7fdf997937ee  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fdf99ddd17b  caffe::Net<>::CopyTrainedLayersFrom()
    @     0x7fdf99de67a5  caffe::Net<>::CopyTrainedLayersFromBinaryProto()
    @     0x7fdf99de683e  caffe::Net<>::CopyTrainedLayersFrom()
    @           0x409208  (unknown)
    @           0x407828  (unknown)
    @     0x7fdf9841c610  __libc_start_main
    @           0x407f99  (unknown)
```

I am not sure what is causing this error because I am actually testing on the same dataset that I am training on so it shouldn't be a dimensionality problem.

**Comment by brollb** [comment]
mentioned in commit 153bb1d1c4ebed9182f4b1d500aedaeee8be8e01

**Comment by brollb** [comment]
Status changed to closed by commit 153bb1d1c4ebed9182f4b1d500aedaeee8be8e01
